encoder-otel-brave/src/test/java/zipkin2/reporter/otel/brave/ITZipkinSenderTests.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    public void close() {
+      braveTraceIdGenerator.close();
+    }
+  }
+
+  /**
+   * Actual testing logic that uses Brave to generate spans and send them to the backend.
+   */
+  static class BraveTraceIdGenerator {
+
+    private final BraveHttpSenderProvider braveHttpSenderProvider = new BraveHttpSenderProvider();
+
+    private final int port;
+
+    Tracing tracing;
+
+    BraveTraceIdGenerator(int port) {
+      this.port = port;
+    }
+
+    List<String> traceIds() throws Exception {
+      ThreadLocalCurrentTraceContext braveCurrentTraceContext = ThreadLocalCurrentTraceContext.newBuilder()
+          .build();
+      List<String> traceIds = new ArrayList<>();
+      tracing = Tracing.newBuilder()
+          .currentTraceContext(braveCurrentTraceContext)
+          .supportsJoin(false)
+          .traceId128Bit(true)
+          .sampler(Sampler.ALWAYS_SAMPLE)
+          .addSpanHandler(braveHttpSenderProvider.apply(port))
+          .localServiceName("my-service")
+          .build();
+      Tracer braveTracer = tracing.tracer();
+
+      for (int i = 0; i < EXPECTED_TRACE_SIZE; i++) {
+        Span span = braveTracer.nextSpan().name("foo " + i)
+            .tag("foo tag", "foo value")
+            .kind(Kind.CONSUMER)
+            .error(new RuntimeException("BOOOOOM!"))
+            .remoteServiceName("remote service")
+            .start();
+        try (SpanInScope scope = braveTracer.withSpanInScope(span)) {
+          String traceId = span.context().traceIdString();
+          log.info("Trace Id <" + traceId + ">");
+          span.remoteIpAndPort("http://localhost", 123456);
+          Thread.sleep(50);
+          span.annotate("boom!");
+          Thread.sleep(50);
+        } finally {
+          span.finish();
+        }
+
+        traceIds.add(span.context().traceIdString());
+      }
+      flush();
+      return traceIds;
+    }
+
+    private void flush() {
+      braveHttpSenderProvider.flush();
+    }
+
+    public void close() {
+      braveHttpSenderProvider.close();
+      tracing.close();
+    }
+
+    /**
+     * Provides a {@link SpanHandler} that uses OKHttp to send spans to a given port.
+     */
+    static class BraveHttpSenderProvider implements Function<Integer, SpanHandler>, Closeable {
+
+      OkHttpSender okHttpSender;
+
+      AsyncReporter<MutableSpan> reporter;
+
+      SpanHandler spanHandler;
+
+      @Override
+      public void close() {
+        if (reporter != null) {
+          reporter.close();
+        }
+        if (okHttpSender != null) {
+          okHttpSender.close();
+        }
+      }
+
+      @Override
+      public SpanHandler apply(Integer port) {
+        okHttpSender = OkHttpSender.newBuilder()
+            .encoding(Encoding.PROTO3)
+            .endpoint("http://localhost:" + port + "/v1/traces")
+            .build();
+        OtelEncoder otelEncoder = new OtelEncoder(Tags.ERROR);
+        reporter = AsyncReporter.builder(okHttpSender).build(otelEncoder);
```

Comment:
why is this using AsyncReporter when it is AsyncSpanHandler that is used for brave?

Reply:
This class is replaced by [ITOtelEncoderTest](https://github.com/openzipkin-contrib/zipkin-otel/pull/7/files#diff-f1614b313ae6e7ebe173556465333def6f893b421571522e9bb496a42ede83c3) and the comment has also been addressed